### PR TITLE
Create view model factory

### PR DESCRIPTION
**Pull request by hodamohammadi**
creating a singleton of view model factory and using it to instantiate view models

### CODE REVIEW

File: conveyor/src/main/java/com/hodamohammadi/conveyor/fragments/SingleChatFragment.kt [hunk 1]
```diff
@@ -21,6 +24,8 @@ class SingleChatFragment : Fragment(), MessageInput.InputListener, MessageInput.
         MessageInput.TypingListener {
 
     private lateinit var messagesAdapter: MessagesListAdapter<IMessage>
+    private lateinit var chatViewModel: ChatViewModel
+
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/fragments/SingleChatFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/22475477cbccc7dcfed1ba20b7902f4eb7c4c21c/conveyor/src/main/java/com/hodamohammadi/conveyor/fragments/SingleChatFragment.kt#L29"></span>Needless blank line(s)</td>
    </tr>
  </tbody>
</table>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ChatViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.hodamohammadi.conveyor.viewmodels
+
+import android.arch.lifecycle.ViewModel
+
+/**
+ * View model for communication between chat UIs.
+ */
+class ChatViewModel: ViewModel() {
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ChatViewModel.kt:8]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/38482a3d1f78ec85be0a809d66e3740ec196c8f0/conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ChatViewModel.kt#L8"></span>Missing spacing before “:”</td>
    </tr>
  </tbody>
</table>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ViewModelFactory.kt [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package com.hodamohammadi.conveyor.viewmodels
+
+import android.arch.lifecycle.ViewModel
+import android.arch.lifecycle.ViewModelProvider
+
+
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ViewModelFactory.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/22475477cbccc7dcfed1ba20b7902f4eb7c4c21c/conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ViewModelFactory.kt#L6"></span>Needless blank line(s)</td>
    </tr>
  </tbody>
</table>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ViewModelFactory.kt [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+/**
+ * singleton of view model factory to create instances of view models.
+ */
+class ViewModelFactory private constructor(): ViewModelProvider.Factory {
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ViewModelFactory.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/22475477cbccc7dcfed1ba20b7902f4eb7c4c21c/conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ViewModelFactory.kt#L10"></span>Missing spacing before “:”</td>
    </tr>
  </tbody>
</table>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ViewModelFactory.kt [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    override fun <T : ViewModel> create(modelClass: Class<T>): T {
+        return if (modelClass.isAssignableFrom(ChatViewModel::class.java)) {
+            ChatViewModel() as T
+        } else {
+            throw IllegalArgumentException("ViewModel Not Found")
+        }
+    }
+
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ViewModelFactory.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/22475477cbccc7dcfed1ba20b7902f4eb7c4c21c/conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ViewModelFactory.kt#L30"></span>Unexpected blank line(s) before “}”</td>
    </tr>
  </tbody>
</table>

**Comment by ReviewerOfCode** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/hodamohammadi/Conveyor/blob/38482a3d1f78ec85be0a809d66e3740ec196c8f0/conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultMessage.kt#L16">conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultMessage.kt#L16</a> - Missing spacing before “:”</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/hodamohammadi/Conveyor/blob/38482a3d1f78ec85be0a809d66e3740ec196c8f0/conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultUser.kt#L13">conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultUser.kt#L13</a> - Missing spacing before “:”</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/hodamohammadi/Conveyor/blob/38482a3d1f78ec85be0a809d66e3740ec196c8f0/conveyor/src/main/java/com/hodamohammadi/conveyor/services/ConveyorNotificationService.kt#L5">conveyor/src/main/java/com/hodamohammadi/conveyor/services/ConveyorNotificationService.kt#L5</a> - Missing spacing before “:”</td>
    </tr>
  </tbody>
</table>

### AndroidLint found issues

#### Warning (2)

| File | Line | Reason |
| ---- | ---- | ------ |
`conveyor/build.gradle` | 12 | Not targeting the latest versions of Android; compatibility modes apply. Consider testing and updating this version. Consult the android.os.Build.VERSION_CODES javadoc for details. 
`conveyor/src/main/AndroidManifest.xml` | 6 | On SDK version 23 and up, your app data will be automatically backed up, and restored on app install. Your GCM regid will not work across restores, so you must ensure that it is excluded from the back-up set. Use the attribute `android:fullBackupContent` to specify an `@xml` resource which configures which files to backup. More info: https://developer.android.com/training/backup/autosyncapi.html 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ChatViewModel.kt (hunk 2 above)
Comment (hodamohammadi) [reply conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ChatViewModel.kt:8]: I’m assuming these are not valid warnings?

File: conveyor/src/main/java/com/hodamohammadi/conveyor/fragments/SingleChatFragment.kt [hunk 6]
```diff
@@ -21,7 +24,8 @@ class SingleChatFragment : Fragment(), MessageInput.InputListener, MessageInput.
         MessageInput.TypingListener {
 
     private lateinit var messagesAdapter: MessagesListAdapter<IMessage>
-
+    private lateinit var chatViewModel: ChatViewModel
+
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/fragments/SingleChatFragment.kt:28]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/38482a3d1f78ec85be0a809d66e3740ec196c8f0/conveyor/src/main/java/com/hodamohammadi/conveyor/fragments/SingleChatFragment.kt#L28"></span>Trailing space(s)</td>
    </tr>
  </tbody>
</table>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ViewModelFactory.kt [hunk 7]
```diff
@@ -0,0 +1,29 @@
+package com.hodamohammadi.conveyor.viewmodels
+
+import android.arch.lifecycle.ViewModel
+import android.arch.lifecycle.ViewModelProvider
+
+/**
+ * singleton of view model factory to create instances of view models.
+ */
+class ViewModelFactory private constructor(): ViewModelProvider.Factory {
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ViewModelFactory.kt:9]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/38482a3d1f78ec85be0a809d66e3740ec196c8f0/conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ViewModelFactory.kt#L9"></span>Missing spacing before “:”</td>
    </tr>
  </tbody>
</table>

**Pull request closed without merging**